### PR TITLE
manager: add custom annotations as param on create/update instances

### PR DESCRIPTION
**Pull request by morpheu**

### CODE REVIEW

File: internal/pkg/rpaas/manager.go [hunk 1]
```diff
@@ -252,6 +262,43 @@ type CertificateData struct {
 	Key         string `json:"key"`
 }
 
+func getAnnotations(params map[string]interface{}) (annotations map[string]string) {
```
Comment (nettoclaudio) [inline internal/pkg/rpaas/manager.go:265]: There are a bunch of validations here. It'd better returning an error in this function in order to make the client notice what happened.

File: internal/pkg/rpaas/manager.go [hunk 2]
```diff
@@ -252,6 +262,43 @@ type CertificateData struct {
...
+	p, found := params["annotations"]
+	if !found {
+		return
+	}
+	err := json.Unmarshal([]byte(p.(string)), &annotations)
+	if err != nil {
+		return
+	}
```
Comment (nettoclaudio) [inline internal/pkg/rpaas/manager.go:273]: Just a niptick. `p.(string)` might panic against a mal-formed/malicious payload. 

```suggestion
	annotationsStr, ok := p.(string)
	if !ok {
		return
	}

	err := json.Unmarshal([]byte(annotationsStr), &annotations)
	if err != nil {
		return
	}
```

**Review by nettoclaudio — APPROVED** [review]
Just requested small fixes but lgtm overall.

File: internal/pkg/rpaas/manager.go (hunk 2 above)
Comment (morpheu) [reply internal/pkg/rpaas/manager.go:273]: Ouch! Thanks, I didn't notice it

File: internal/pkg/rpaas/manager.go (hunk 1 above)
Comment (morpheu) [reply internal/pkg/rpaas/manager.go]: You're right. At first glance seems to be a first idea just try to ignore them but it's better return it to client

**Pull request merged**